src/libraries/PersonalSignLib.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+library PersonalSignLib {
+    bytes private constant PERSONAL_SIGN_TYPE = "PersonalSign(bytes prefixed)";
+    bytes32 private constant PERSONAL_SIGN_TYPEHASH = keccak256(PERSONAL_SIGN_TYPE);
+
+    /// @notice We don't care how the hash was computed for personal sign, and it does not match the typestring above
+    /// i.e.  keccak256("\x19Ethereum Signed Message:\n" || len(someMessage) || someMessage)
+    function hash(bytes32 message) internal pure returns (bytes32) {
+        return keccak256(abi.encode(PERSONAL_SIGN_TYPEHASH, message));
```

Comment:
hm this seems wrong? the personal sign type defines the input as a bytes param and here it is already a bytes32?

Reply:
Yep I asked viem to see what the reasoning is

Reply:
Can we bring this up with Spearbit bc this still doesn't make sense to me why this is allowed?